functions/src/modules/SQR.functions.ts

```diff
@@ -512,4 +512,16 @@ export const exportSubmissionsToSpreadsheet = functions.database
         'Email Address': withDefault(author.emailAddress),
       });
     }
-  );
+  )
+
+export const processMarkingSubmissionAsDone = async (processObject: any) => {
+  const listName = processObject.fileName.split("-")[0];
+  const soundQualityReporting = db.ref(`files/${listName}/${processObject.fileName}/soundQualityReporting`);
+  const sqrResults = await soundQualityReporting.once('value');
+  if (sqrResults.exists()) {
```

Comment:
I'm not sure how else to write this. Without a variable, we could write like this?

fileRef.once('value').then(results => {
  if (results.exists()) {
    await fileRef.update({
      status: "Done",
      timestampDone: processObject.lastThreadMessageTimestamp,
    })
  }
})

Reply:
I was thinking of something like `if ((await soundQualityReporting.once('value')).exists())`. Not insisting here though.

Reply:
I am not sure if that would work. Since soundQualityReporting.once('value') returns a promise that needs to be awaited. Then the resolved value is what gets called with .exists().

Anyways, I can give it a try and see if there's any errors.